ztag/test/encoding_test.py

```diff
@@ -0,0 +1,96 @@
+            # Host appears twice
+            "Got connection from researchscan{worker}.eecs.umich.edu, rejecting researchscan{worker}.eecs.umich.edu.": "Got connection from CLIENT_HOSTNAME, rejecting CLIENT_HOSTNAME.",
+            # IP appears twice on different lines
+            "Got connection from 141.212.{ip3}.{ip4}\nrejecting 141.212.{ip3}.{ip4}.": "Got connection from CLIENT_IP\nrejecting CLIENT_IP.",
+            # Host appears twice on different lines
+            "Got connection from researchscan{worker}.eecs.umich.edu\nrejecting researchscan{worker}.eecs.umich.edu.": "Got connection from CLIENT_HOSTNAME\nrejecting CLIENT_HOSTNAME.",
+            # SFJ: Both IP and host
+            "Got connection from 198.108.66.{ip4}:1234 (worker-{worker}.sfj.corp.censys.io:1234)": "Got connection from CLIENT_IP:1234 (CLIENT_HOSTNAME:1234)",
+            # SFJ: IP at end
+            "Got connection from 198.108.66.{ip4}": "Got connection from CLIENT_IP",
+            # SFJ: IP at start
+            "198.108.66.{ip4} connected.": "CLIENT_IP connected.",
+            # SFJ: Host at end
+            "Got connection from worker-{worker}.sfj.corp.censys.io": "Got connection from CLIENT_HOSTNAME",
+            # SFJ: Host at start
+            "worker-{worker}.sfj.corp.censys.io connected.": "CLIENT_HOSTNAME connected.",
+            # SFJ: Isolated IP
+            "198.108.66.{ip4}": "CLIENT_IP",
+            # SFJ: Isolated host
+            "worker-{worker}.sfj.corp.censys.io": "CLIENT_HOSTNAME",
+            # SFJ: IP appears twice
+            "Got connection from 198.108.66.{ip4}, rejecting 198.108.66.{ip4}.": "Got connection from CLIENT_IP, rejecting CLIENT_IP.",
+            # SFJ: Host appears twice
+            "Got connection from worker-{worker}.sfj.corp.censys.io, rejecting worker-{worker}.sfj.corp.censys.io.": "Got connection from CLIENT_HOSTNAME, rejecting CLIENT_HOSTNAME.",
+            # SFJ: IP appears twice on different lines
+            "Got connection from 198.108.66.{ip4}\nrejecting 198.108.66.{ip4}.": "Got connection from CLIENT_IP\nrejecting CLIENT_IP.",
+            # SFJ: Host appears twice on different lines
+            "Got connection from worker-{worker}.sfj.corp.censys.io\nrejecting worker-{worker}.sfj.corp.censys.io.": "Got connection from CLIENT_HOSTNAME\nrejecting CLIENT_HOSTNAME."
+
+        }
+        cases = {}
+        for k, v in cases_format.items():
+            if not v:
+                cases_format[k] = k
+                v = k
+            temp = v
+            for ip3 in (121, 122):
+                for ip4 in range(0, 256):
+                    for worker in range(1, 10):
+                        kk = k.format(worker=worker, ip3=ip3, ip4=ip4)
+                        vv = v.format(worker=worker, ip3=ip3, ip4=ip4)
+                        cases[kk] = vv
+
+        for input, expected in cases.items():
```

Comment:
This overwrites built-in `input`.

Reply:
Good old python -- making the assumption that no one would use obscure variable names like `input`, `file` or `dir`.

Reply:
> Good old python -- making the assumption that no one would use obscure variable names like `input`, `file` or `dir`.

![The more you know!](https://user-images.githubusercontent.com/14554/44864649-2bf98700-ac4e-11e8-9e00-af74cbfa4388.gif)